tutorials/quickstart/tutorial.md

```diff
@@ -123,9 +123,105 @@ NOTE: If you forgot to detach the debugger, you'll need to click the **Continue*
+
+So what happened here? The message couldn't be successfully processed during an immediate round of retries, so it delayed the message for 10 seconds to try again. After that, it could still not be processed successfully, so it delayed the message for an additional 20 seconds, and then 30 seconds, before giving up all hope and transferring the message to an **error queue**, a holding location for poison messages so that other messages behind it can still get processed successfully.
+
+Once the message enters the error queue, ServicePulse takes over, displaying all failed messages grouped by exception type and the location where it's thrown from.
+
+If you click anywhere on the exception group, it will take you to the list of exceptions within that group. This is not too interesting, since we currently only have one, but if you click again on the individual exception, you will get a rich exception detail view:
+
+![ServicePulse: Exception Details](pulse-exception-details.png)
+
+No need to go digging through log files to find out what went wrong. ServicePulse provides the exception's stack trace, message headers, and message body right here.
+
+Armed with this information, it should be much easier to track down and fix our bug, so let's do that:
+
+1. Because we ran without attaching the debugger, we need to close both browser windows and all console applications.
+1. In the **Sales** endpoint, locate and open the **PlaceOrderHandler.cs** file.
+1. Comment out the `throw` statement, and uncomment all the code below the **ThrowFatalException** region, returning the code to its original working state.
+1. Start the solution again. It won't be throwing any exceptions so attaching the debugger is fine this time.
+2. Once the **ServicePulse** window launches, navigate to the **Failed Messages** view.
+
+Now our system has been fixed, and we can give that failed message another chance.
+
+1. Move the **Sales** and **Billing** windows around so you can see waht happens when you retry the message.
+2. In the **ServicePulse** window, click the **Request Retry** link.
+3. In the confirmation dialog, click the **Yes** button, and remember to watch the **Sales** and **Billing** windows.
+4. It may take several seconds to enqueue the batch, but eventually you will see the familiar log messages in **Sales** and **Billing**, showing the message being processed successfully as if nothing bad ever happened.
+
+This is a powerful feature. Many systemic failures are the result of bad deployments. A new version is rolled out with a bug, and errors suddenly start appearing that ultimately result in lost data.
+
+With a message-based system, no data is ever lost, because those failures result in messages being sent to an error queue, not lost to the ether. After a deploy, you can watch ServicePulse, and if messages start to pile up in the error queue, you can revert to the previous known good configuration while you diagnose the problem.
```

Comment:
```suggestion
With a message-based system, no data is ever lost, because those failures result in messages being sent to an error queue, not lost to the ether. After a deployment, you can watch ServicePulse, and if messages start to pile up in the error queue, you can revert to the previous known good configuration while you diagnose the problem.
```